src/components/Link/Link.test.tsx

```diff
@@ -1,19 +1,150 @@
+    const size = faker.helpers.arrayElement(
+      Object.keys(fontSizes)
+    ) as keyof typeof fontSizes;
 
-test.todo("adds rel='noopener noreferrer' when target is '_blank'");
+    render(<Basic size={size} />);
+    expect(screen.getByRole("link")).toHaveStyle({ fontSize: size });
+  });
 
-test.todo("can be used with Icon component");
+  test("applies the correct font weight based on the 'weight' prop", () => {
+    const weight = faker.helpers.arrayElement(
+      Object.keys(fontWeights)
+    ) as keyof typeof fontWeights;
+
+    render(<Basic weight={weight} />);
+    expect(screen.getByRole("link")).toHaveClass(`fw_${weight}`);
+  });
+
+  test("applies the correct styles for low and high contrast", () => {
+    render(<Contrasts />);
+
+    expect(screen.getByRole("link", { name: "낮은 명암비" })).toHaveClass(
+      "c_text.muted"
+    );
+
+    expect(screen.getByRole("link", { name: "높은 명암비" })).toHaveClass(
+      "c_text"
+    );
+  });
+
+  test("applies the correct underline styles", () => {
+    render(<Underlines />);
+
+    expect(screen.getByRole("link", { name: "밑줄 있음" })).toHaveClass(
+      "td_underline"
+    );
+    expect(screen.getByRole("link", { name: "밑줄 없음" })).toHaveClass(
+      "td_none"
+    );
+  });
+});
+
+describe("behavior test", () => {
+  test("forwards additional anchor props like 'href' and 'target'", () => {
+    const href = faker.internet.url({ appendSlash: true });
+    const target = "_self";
+
+    render(<Basic href={href} target={target} />);
+
+    const link = screen.getByRole("link");
+    expect(link).toHaveAttribute("href", href);
+    expect(link).toHaveAttribute("target", target);
+  });
+
+  test("adds rel='noopener noreferrer' when target is '_blank'", () => {
+    render(<Security />);
+
+    expect(
+      screen.getByRole("link", { name: "새 탭에서 열기 (보안 속성 자동 추가)" })
+    ).toHaveAttribute("rel", "noopener noreferrer");
+
+    expect(
+      screen.getByRole("link", { name: "같은 탭에서 열기" })
+    ).not.toHaveAttribute("rel", "noopener noreferrer");
+  });
+
+  test("navigates to the correct URL when clicked", async () => {
+    const href = faker.internet.url({ appendSlash: true });
+    render(<Basic href={href} />);
+
+    const link = screen.getByRole("link");
+    expect(link).toHaveAttribute("href", href);
+    await userEvent.click(link);
+
+    expect(window.location.href).toBe(href);
+  });
+
+  test("handles click events correctly", async () => {
+    const handleClick = vi.fn();
+    render(<Basic href="#" onClick={handleClick} />);
+
+    const link = screen.getByRole("link");
+    await userEvent.click(link);
+
+    expect(handleClick).toHaveBeenCalledTimes(1);
+  });
+
+  test("link is keyboard focusable", async () => {
+    const href = faker.internet.url({ appendSlash: true });
+    render(<Basic href={href} />);
+
+    const link = screen.getByRole("link");
+    await userEvent.tab();
+    expect(link).toHaveFocus();
```

Comment:
nit: 사용자의 의식의 흐름 상 이 순서가 더 자연스럽지 않을까요?

```suggestion
    await userEvent.tab();
    
    const link = screen.getByRole("link");
    expect(link).toHaveFocus();
```

Reply:
맞습니다. 사용자 관점에서 작성해야하는데 무의식적으로 변수를 위로 올려야한다고 생각한거 같습니다. 수정하도록 하겠습니다.